assets/src/components/d3/createResourceAccessChart.js

```diff
@@ -66,36 +65,78 @@ function appendLegend (svg) {
 
   legendRect
     .attr('y', (_, i) => legendY + i * legendInterval)
-    .attr('x', w)
+    .attr('x', 0)
     .style('fill', d => d[1])
 }
 
+function truncate (selection, labelWidth) {
+  selection.each(function (d) {
+    const textEl = d3.select(this)
+    const textString = d.split('|')[1]
+    textEl.text(textString)
+    const textPixelLength = textEl.node().getComputedTextLength()
+    if (textPixelLength > labelWidth) {
+      const numChars = Math.floor(labelWidth / textPixelLength * textString.length) - 3
+      textEl.text(`${textString.substring(0, numChars - 1)}...`)
+    } else {
+      textEl.text(textString)
+    }
+  })
+}
+
 function createResourceAccessChart ({ data, width, height, domElement }) {
   const resourceData = data.sort((a, b) => b.total_percent - a.total_percent)
 
-  const [mainWidth, miniHeight] = adjustViewport(width, height, mainMargin)
+  const sideMarginSize = width * 0.075
+  const margin = { top: 0, right: sideMarginSize, bottom: 0, left: sideMarginSize }
+
+  const [availWidth, availHeight] = adjustViewport(width, height, margin)
+
+  /*
+  The decimal multipliers for mainMargin.left, mainWidth, mainMargin.right, miniMargin.left,
+  miniWidth, and miniMargin.right should add up to 1.0. In other words, the sum of those values
+  should be equal to the availWidth.
```

Comment:
I'm guessing this comment is also applicable to availHeight?

Reply:
Yeah, I will update that. Thanks!